src/ansiblelint/cli.py

```diff
@@ -196,10 +228,20 @@ def get_cli_parser() -> argparse.ArgumentParser:
     )
     parser.add_argument(
         "--write",
-        dest="write",
-        action="store_true",
-        help="Reformat YAML files to standardize spacing, quotes, etc. "
-        "Future versions will expand this option so it fixes more issues.",
+        dest="write_list",
+        # this is a tri-state argument that takes an optional comma separated list:
+        #   not provided, --write, --write=a,b,c
+        const="none",  # --write (no option) implicitly stores this
+        nargs="?",  # either 0 (--write) or 1 (--write=a,b,c) argument
+        action=WriteArgAction,
+        help="Allow ansible-lint to modify files. "
+        "'--write' will reformat YAML files to standardize spacing, quotes, etc. "
+        "'--write=all' will also run all available rule transforms on files "
+        "(A rule transform can fix or simplify fixing issues identified by that rule). "
+        "'--write=rule-tag,rule-id' will limit running transforms to rules that match "
+        "the given comma-separated list of rule ids or rule tags. "
+        "'--write=none' behaves the same as '--write' to reformat without running "
+        "transforms.",
```

Comment:
Should `--write` be equivalent to `--write=none` or `--write=all`?

Reply:
Write without arguments should behave as `all`.

Reply:
```suggestion
        help="Allow ansible-lint to modify files. "
        "'--write' will reformat YAML files to standardize spacing, quotes, etc. "
        "''--write' will also run all available rule transforms on files "
        "(A rule transform can fix or simplify fixing issues identified by that rule). "
        "'--write=all' behaves the same as '--write' to reformat and run all transforms."
        "'--write=none' reformats without running any of the rule transforms."
        "'--write=rule-tag,rule-id' will reformat and run a subset of transforms.
        "The transforms subset includes rules that match "
        "the given comma-separated list of rule ids or rule tags. "
```

Reply:
done.